static/app/views/performance/transactionSummary/transactionOverview/durationChart/content.tsx

```diff
@@ -56,6 +60,44 @@ function Content({
     );
   }
 
+  const colors = (data && theme.charts.getColorPalette(data.length - 2)) || [];
+
+  // Create a list of series based on the order of the fields,
+  // We need to flip it at the end to ensure the series stack right.
+  const series = data
+    ? data
+        .map((values, i: number) => {
+          return {
+            ...values,
+            color: colors[i],
+            lineStyle: {
+              opacity: 0,
+            },
+          };
+        })
+        .reverse()
+    : [];
+
+  // Finds the range of yAxis values based on what is selected in the legend
+  let range;
+  if (series[1]) {
+    let minSeries = series[0];
+    let maxSeries;
+    series.forEach(({seriesName}, idx) => {
+      if (legend?.selected?.[seriesName] !== false) {
+        minSeries = series[idx];
+        maxSeries ??= series[idx];
```

Comment:
how would this work if there are > 2 series. wouldn't then `maxSeries` always be `series[0]` and then `minSeries` be `series[last series]`?

Reply:
yeah let's move this to utils and unit test this so it's clear how this function should behave when there are different inputs

Reply:
What it should do (if my logic isn't messed, i tested and it seemed good) is select the series with the smallest and largest index, that are selected in the legend. So in the case where all series are selected in the legend, then yes `maxSeries` always be `series[0]` and then minSeries be `series[last series]`.

This makes sense because series[0] corresponds to p100, series[1] -> p99, etc 
and p100 > p99 > p95 > ....

Reply:
> yeah let's move this to utils and unit test this so it's clear how this function should behave when there are different inputs

@udameli Extracted to utils and added the unit tests :)